openstack/baremetal/v1/nodes/requests.go

```diff
@@ -478,3 +479,89 @@ func ChangePowerState(client *gophercloud.ServiceClient, id string, opts PowerSt
 	})
 	return
 }
+
+// This is the desired RAID configuration on the bare metal node.
+type TargetRAIDConfig struct {
+	LogicalDisks []LogicalDisk `json:"logical_disks"`
+}
+
+// TargetRAIDConfigBuilder allows extensions to modify a set RAID config request.
+type TargetRAIDConfigBuilder interface {
+	ToRAIDConfigMap() (map[string]interface{}, error)
+}
+
+type LogicalDisk struct {
+	// Size (Integer) of the logical disk to be created in GiB.  If unspecified, "MAX" will be used.
+	SizeGB *int `json:"size_gb"`
+
+	// RAID level for the logical disk.
+	RAIDLevel string `json:"raid_level" required:"true"`
+
+	// Name of the volume. Should be unique within the Node. If not specified, volume name will be auto-generated.
+	VolumeName string `json:"volume_name,omitempty"`
+
+	// Set to true if this is the root volume. At most one logical disk can have this set to true.
+	IsRootVolume *bool `json:"is_root_volume,omitempty"`
+
+	// Set to true if this logical disk can share physical disks with other logical disks.
+	SharePhysicalDisks *bool `json:"share_physical_disks,omitempty"`
+
+	// If this is not specified, disk type will not be a criterion to find backing physical disks
+	DiskType string `json:"disk_type,omitempty"`
+
+	// If this is not specified, interface type will not be a criterion to find backing physical disks.
+	InterfaceType string `json:"interface_type,omitempty"`
+
+	// Integer, number of disks to use for the logical disk. Defaults to minimum number of disks required
+	// for the particular RAID level.
+	NumberOfPhysicalDisks int `json:"number_of_physical_disks,omitempty"`
+
+	// The name of the controller as read by the RAID interface.
+	Controller string `json:"controller,omitempty"`
+
+	// A list of physical disks to use as read by the RAID interface.
+	PhysicalDisks []map[string]interface{} `json:"physical_disks,omitempty"`
+}
+
+// Custom marshal function for LogicalDisk to send "MAX" if SizeGB isn't specified.
+func (disk *LogicalDisk) MarshalJSON() ([]byte, error) {
+	type Alias LogicalDisk
+
+	var sizeGB interface{}
+	if disk.SizeGB == nil {
+		sizeGB = "MAX"
+	} else {
+		sizeGB = disk.SizeGB
+	}
+
+	return json.Marshal(&struct {
+		SizeGB interface{} `json:"size_gb"`
+		*Alias
+	}{
+		SizeGB: sizeGB,
+		Alias:  (*Alias)(disk),
+	})
+}
+
+func (t *TargetRAIDConfig) ToRAIDConfigMap() (map[string]interface{}, error) {
+	body, err := gophercloud.BuildRequestBody(t, "")
+	if err != nil {
+		return nil, err
+	}
+
+	return body, nil
+}
+
+// Request to change a Node's RAID config.
+func SetRAIDConfig(client *gophercloud.ServiceClient, id string, targetRAIDConfig TargetRAIDConfig) (r ChangeStateResult) {
```

Comment:
`s/TargetRAIDConfig/RAIDConfigOptsBuilder/`

Two changes: the name of the builder and that you want to use the builder (not the struct) as the type. This allows other structs conforming to the builder to be passed in.